src/store/LocalStorageStore.ts

```diff
@@ -0,0 +1,207 @@
+import {CommentData, EntityType, FileData, ProjectData, SectionData} from '../types';
+
+export interface LocalStorageDependency {
+    getItem(key: string): string | null;
+    setItem(key: string, value: string): void;
+    clear(): void;
+}
+
+// TODO: versioning of the store allows for migrations
+// const LOCAL_STORAGE_KEY_VERSION = 'version';
+
+const LOCAL_STORAGE_KEY_PROJECTS = 'projects';
+const LOCAL_STORAGE_KEY_SECTIONS = 'sections';
+const LOCAL_STORAGE_KEY_FILES = 'files';
+const LOCAL_STORAGE_KEY_COMMENTS = 'comments';
+
+export type StoreData = {
+    projects: ProjectData[];
+    sections: SectionData[];
+    files: FileData[];
+    comments: CommentData[];
+}
+
+export class LocalStorageStore {
+    private ls: LocalStorageDependency;
+
+    private currentData: StoreData;
+
+    constructor(ls: LocalStorageDependency) {
+        this.ls = ls;
+        this.currentData = this.migrateLocalStorageStore();
+
+        if (!this.currentData.projects.length) {
+            this.initializeWithSampleData();
+        }
+
+        // this.clear();
+    }
+
+    clear = () => {
+        this.ls.clear();
+        this.currentData = this.migrateLocalStorageStore();
+    }
+
+    initializeWithSampleData = () => {
+        this.clear();
+        this.setAllProjects(initialProjects);
+        this.setAllSections(initialSections);
+        this.setAllFiles(initialFiles);
+        this.setAllComments(initialComments);
+    }
```

Comment:
I'm thinking we can have two buttons on the UI:

- "Clear data" - Calls `store.clear()` and reloads the page
- "Reset with sample data" - Calls `store.initializeWithSampleData()` and reloads the page

The reloading of the page allows the application to re-initialize from local storage appropriately

Reply:
Very interesting. I'd love to know more about your thought process for having those be the two buttons. Especially from a data flow perspective

Reply:
They're just for debugging really. Those buttons would not be present in the production version of the application. The data flow is "update local storage then restart the program through reloading the page". That's the least obtrusive way to put that debug functionality in there I think. No special case code that's like "seed this data then re-read from local storage" anywhere in the code

Reply:
Not sure I understand your question fully though